src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,65 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
 public class WorkWithFile {
+    private static final char COMMA = ',';
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        int sumOfSupply = 0;
+        int sumOfBuy = 0;
+        int valueAfterComma;
+
+        File fromFile = new File(fromFileName);
+        StringBuilder stringBuilder = new StringBuilder();
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile));
+            String readDataFromFile = bufferedReader.readLine();
+            while (readDataFromFile != null) {
+                stringBuilder.append(readDataFromFile).append(" ");
+                readDataFromFile = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read file: " + fromFileName, e);
+        }
+
+        String[] dataFromFile = stringBuilder.toString().split(" ");
+        for (String data: dataFromFile) {
+            valueAfterComma = Integer.parseInt(data.substring(data.indexOf(COMMA) + 1));
+            if (data.startsWith(SUPPLY)) {
+                sumOfSupply += valueAfterComma;
+            } else {
+                sumOfBuy += valueAfterComma;
+            }
+        }
+        String[] dataWriteToFile = new String[3];
+        dataWriteToFile[0] = SUPPLY + COMMA + sumOfSupply + System.lineSeparator();
+        dataWriteToFile[1] = BUY + COMMA + sumOfBuy + System.lineSeparator();
+        dataWriteToFile[2] = RESULT + COMMA + (sumOfSupply - sumOfBuy);
```

Comment:
Let's extract it to the separate `private` method. Remember from `checklist`:

**Divide program logic.** For example: “write to file” or “create report” - are a separate operations. So we should create separate methods for them.

Also, let's build a single result String, not the array.